Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs

```diff
@@ -0,0 +1,119 @@
+        /// The _logger
+        /// </summary>
+        private readonly ILogger _logger;
+
+        private readonly IInstallationManager _installationManager;
+
+        private readonly IApplicationHost _appHost;
+
+        public PluginUpdateTask(ILogger logger, IInstallationManager installationManager, IApplicationHost appHost)
+        {
+            _logger = logger;
+            _installationManager = installationManager;
+            _appHost = appHost;
+        }
+
+        /// <summary>
+        /// Creates the triggers that define when the task will run
+        /// </summary>
+        /// <returns>IEnumerable{BaseTaskTrigger}.</returns>
+        public IEnumerable<TaskTriggerInfo> GetDefaultTriggers()
+        {
+            return new[] { 
+            
+                // At startup
+                new TaskTriggerInfo {Type = TaskTriggerInfo.TriggerStartup},
+
+                // Every so often
+                new TaskTriggerInfo { Type = TaskTriggerInfo.TriggerInterval, IntervalTicks = TimeSpan.FromHours(24).Ticks}
+            };
+        }
+
+        /// <summary>
+        /// Update installed plugins
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <param name="progress">The progress.</param>
+        /// <returns>Task.</returns>
+        public async Task Execute(CancellationToken cancellationToken, IProgress<double> progress)
+        {
+            progress.Report(0);
+
+            var packagesToInstall = (await _installationManager.GetAvailablePluginUpdates(typeof(PluginUpdateTask).Assembly.GetName().Version, true, cancellationToken).ConfigureAwait(false)).ToList();
+
+            progress.Report(10);
+
+            var numComplete = 0;
+
+            foreach (var package in packagesToInstall)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    await _installationManager.InstallPackage(package, true, new SimpleProgress<double>(), cancellationToken).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    // InstallPackage has it's own inner cancellation token, so only throw this if it's ours
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        throw;
+                    }
+                }
+                catch (HttpException ex)
+                {
+                    _logger.LogError(ex, "Error downloading {0}", package.name);
+                }
+                catch (IOException ex)
+                {
+                    _logger.LogError(ex, "Error updating {0}", package.name);
+                }
+
+                // Update progress
+                lock (progress)
+                {
+                    numComplete++;
+                    progress.Report(90 * numComplete / packagesToInstall.Count + 10);
```

Comment:
You've missed a dot, this would fail on rounding issues (right @Bond-009? 😄 )
```suggestion
                    progress.Report(90.0 * numComplete / packagesToInstall.Count + 10);
```